src/grasp_generator.cpp

```diff
@@ -658,50 +668,42 @@ bool GraspGenerator::addGrasp(const Eigen::Affine3d& grasp_pose, const GraspData
                               std::vector<GraspCandidatePtr>& grasp_candidates, const Eigen::Affine3d& object_pose,
                               double object_width)
 {
-  if (verbose_)
+  if (show_grasp_arrows_)
+    visual_tools_->publishZArrow(grasp_pose, rviz_visual_tools::BLUE, rviz_visual_tools::XSMALL, 0.02);
+  if (show_grasp_axis_)
+    visual_tools_->publishAxis(grasp_pose, 0.02, 0.002);
+  if (show_grasp_axis_ || show_grasp_arrows_)
   {
-    // visual_tools_->publishAxis(grasp_pose, 0.02, 0.002);
-    visual_tools_->publishZArrow(grasp_pose, rviz_visual_tools::BLUE, rviz_visual_tools::XXSMALL, 0.01);
-    ros::Duration(0.01).sleep();
+    visual_tools_->trigger();
+    ros::Duration(show_grasp_arrows_speed_).sleep();
   }
 
   // The new grasp
   moveit_msgs::Grasp new_grasp;
 
   // Approach and retreat
   // aligned with pose (aligned with grasp pose z-axis
-  // TODO:: Currently the pre/post approach/retreat translations are not robot agnostic.
   // It currently being loaded with the assumption that z-axis is pointing away from object.
 
   // set pregrasp
-  moveit_msgs::GripperTranslation pre_grasp_approach;
   new_grasp.pre_grasp_approach.direction.header.stamp = ros::Time::now();
-  new_grasp.pre_grasp_approach.desired_distance =
+  new_grasp.pre_grasp_approach.desired_distance = 
       grasp_data->finger_to_palm_depth_ + grasp_data->approach_distance_desired_;
-  new_grasp.pre_grasp_approach.min_distance = 0;  // NOT IMPLEMENTED
-  new_grasp.pre_grasp_approach.direction.header.frame_id = grasp_data->parent_link_->getName();
-  new_grasp.pre_grasp_approach.direction.vector.x = 0;
-  new_grasp.pre_grasp_approach.direction.vector.y = 0;
-  new_grasp.pre_grasp_approach.direction.vector.z = -1;
-  // new_grasp.pre_grasp_approach.direction.header.frame_id = "world";
-  // new_grasp.pre_grasp_approach.direction.vector.x = 1;
-  // new_grasp.pre_grasp_approach.direction.vector.y = 0;
-  // new_grasp.pre_grasp_approach.direction.vector.z = 0;
+  new_grasp.pre_grasp_approach.min_distance = new_grasp.pre_grasp_approach.desired_distance / 2;  // NOT IMPLEMENTED
```

Comment:
I didn't really implement it, I am actually not sure what the min distance does. Please tell me. I set it to that value because if it was same as desired it would fail.